plugins/nf-schema/src/main/nextflow/validation/SamplesheetConverter.groovy

```diff
@@ -19,16 +19,10 @@ import nextflow.Nextflow
 @CompileStatic
 class SamplesheetConverter {
 
-    private static Path samplesheetFile
-    private static Path schemaFile
     private static ValidationConfig config
-    private static Map options
 
-    SamplesheetConverter(Path samplesheetFile, Path schemaFile, ValidationConfig config, Map options) {
-        this.samplesheetFile = samplesheetFile
-        this.schemaFile = schemaFile
+    SamplesheetConverter(ValidationConfig config) {
         this.config = config
-        this.options = options
     }
```

Comment:
Mainly this. The samplesheet file and schema file were being wrongly cashed when they were class factory inputs. Making them function inputs doesn't cache them so they are always the correct values (I've no idea why but it works)